typescript.js

```diff
@@ -0,0 +1,329 @@
+    // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md
+    // 'import/no-extraneous-dependencies': [],
+};
+
+module.exports = {
+    extends: [
+        'airbnb',
+        'airbnb/hooks',
+        'plugin:@typescript-eslint/recommended',
+        'prettier',
+    ],
+    parser: '@typescript-eslint/parser',
+    settings: {
+        // Apply special parsing for TypeScript files
+        'import/parsers': {
+            '@typescript-eslint/parser': ['.ts', '.tsx', '.d.ts'],
+        },
+        // Append 'ts', 'tsx' extensions to Airbnb 'import/resolver' setting
+        'import/resolver': {
+            node: {
+                extensions: ['.js', '.json', '.ts', '.tsx', '.d.ts'],
+            },
+        },
+        // Append 'ts' extensions to Airbnb 'import/extensions' setting
+        'import/extensions': ['.js', '.mjs', '.ts', '.tsx', '.d.ts'],
+        // Resolve type definition packages
+        'import/external-module-folders': [
+            'node_modules',
+            'node_modules/@types',
+        ],
+    },
+    env: {
+        ...base.env,
+    },
+    rules: {
+        ...base.rules,
+        ...airbnbTypeScript,
+        'import/extensions': [
+            2,
+            'always',
+            {
+                ignorePackages: true,
+                pattern: {
+                    js: 'never',
+                    ts: 'never',
+                    tsx: 'never',
+                },
+            },
+        ],
+        'react/jsx-filename-extension': [2, { extensions: ['.js', '.tsx'] }],
+        // Disabled so we can lint Node files (Webpack, Vite, Jest) more easily.
+        '@typescript-eslint/no-var-requires': 0,
+    },
+    overrides: [
+        {
+            files: ['*.ts', '*.tsx'],
+            rules: {
+                // This is not needed in TypeScript, where `defaultProps` aren’t used,
+                // and function default parameters are introspect-able by the type checker.
+                'react/require-default-props': 0,
+                // Copy from eslint-config-airbnb-typescript
+                // https://github.com/iamturns/eslint-config-airbnb-typescript/blob/d30b7d2a41f3f2cf899cad3ed69e3b7eacbc7234/lib/shared.js#L256
+                // The following rules are enabled in Airbnb config, but are already checked (more thoroughly) by the TypeScript compiler
+                // Some of the rules also fail in TypeScript files, for example: https://github.com/typescript-eslint/typescript-eslint/issues/662#issuecomment-507081586
+                'constructor-super': 'off',
+                'getter-return': 'off',
+                'no-const-assign': 'off',
+                'no-dupe-args': 'off',
+                'no-dupe-class-members': 'off',
+                'no-dupe-keys': 'off',
+                'no-func-assign': 'off',
+                'no-new-symbol': 'off',
+                'no-obj-calls': 'off',
+                'no-redeclare': 'off',
```

Comment:
In line 149 we deactivate `no-redeclare` then define the typescript rule. Is this override necessary?

Reply:
Likely not! I’ll edit, I think we should also report this to eslint-config-airbnb-typescript.